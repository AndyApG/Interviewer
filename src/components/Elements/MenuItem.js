import React from 'react';
import { Link } from 'react-router';

function MenuItem(props) {
  return (
    <Link to={props.to} onClick={props.onClick} activeClassName='bm-item-active'>
      {props.imageType && <img src={"/images/"+props.imageType+".svg"} />}
      {props.title}
    </Link>
  );
}

export default MenuItem;
