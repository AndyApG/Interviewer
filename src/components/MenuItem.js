import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Icon } from '../ui/components';

/**
  * Renders a menu item. Image is based on icon.
  */
function MenuItem(props) {
  const { icon, interfaceType, isActive, label, menuType, onClick, to } = props;

  const iconMap = {
    NameGenerator: 'menu-name-generator',
    NameGeneratorAutoComplete: 'menu-name-generator',
    NameGeneratorList: 'menu-name-generator',
    Sociogram: 'menu-sociogram',
  };

  const mappedIcon = icon || (interfaceType && iconMap[interfaceType]) || 'menu-default-interface';

  const itemClasses = classNames(
    'menu__menuitem',
    {
      'menu__menuitem--active': isActive,
    },
    `menu__menuitem--${menuType}`,
  );

  return (
    to ? (
      <NavLink
        onClick={onClick}
        className={itemClasses}
        activeClassName="menu__menuitem--active"
        to={to}
      >
        <Icon name={mappedIcon} />
        {label}
      </NavLink>) :
      (<a onClick={onClick} className={itemClasses} tabIndex={0} role="menuitem">
        <Icon name={mappedIcon} />
        {label}
      </a>)
  );
}

MenuItem.propTypes = {
  icon: PropTypes.string,
  interfaceType: PropTypes.string,
  isActive: PropTypes.bool,
  label: PropTypes.string.isRequired,
  menuType: PropTypes.string,
  onClick: PropTypes.func,
  to: PropTypes.string,
};

MenuItem.defaultProps = {
  icon: '',
  interfaceType: '',
  isActive: false,
  menuType: 'primary',
  onClick: () => {},
  to: '',
};

export default MenuItem;
