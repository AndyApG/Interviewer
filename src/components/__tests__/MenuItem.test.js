/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import MenuItem from '../../Elements/MenuItem';

const toLink = 'linkyLink';
const onClick = () => {};
const imageType = 'type';
const title = 'title';

describe('MenuItem component', () => {
  it('renders menu item with label', () => {
    const component = shallow(
      <MenuItem to={toLink} onClick={onClick} imageType={imageType} label={title} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('renders menu item with no image', () => {
    const component = shallow(
      <MenuItem to={toLink} onClick={onClick} label={title} />,
    );

    expect(component).toMatchSnapshot();
  });
});
