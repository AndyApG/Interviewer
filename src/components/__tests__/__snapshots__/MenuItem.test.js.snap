// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component renders menu item with label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    icon=""
    imageType="type"
    interfaceType=""
    isActive={false}
    label="title"
    menuType="primary"
    onClick={[Function]}
    to="linkyLink"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "menu__menuitem--active",
      "ariaCurrent": "true",
      "children": Array [
        <Icon
          className=""
          color=""
          name="menu-default-interface"
          style={Object {}}
/>,
        "title",
      ],
      "className": "menu__menuitem menu__menuitem--primary",
      "onClick": [Function],
      "to": "linkyLink",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "color": "",
          "name": "menu-default-interface",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "title",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "menu__menuitem--active",
        "ariaCurrent": "true",
        "children": Array [
          <Icon
            className=""
            color=""
            name="menu-default-interface"
            style={Object {}}
/>,
          "title",
        ],
        "className": "menu__menuitem menu__menuitem--primary",
        "onClick": [Function],
        "to": "linkyLink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "",
            "name": "menu-default-interface",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "title",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MenuItem component renders menu item with no image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    icon=""
    interfaceType=""
    isActive={false}
    label="title"
    menuType="primary"
    onClick={[Function]}
    to="linkyLink"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "menu__menuitem--active",
      "ariaCurrent": "true",
      "children": Array [
        <Icon
          className=""
          color=""
          name="menu-default-interface"
          style={Object {}}
/>,
        "title",
      ],
      "className": "menu__menuitem menu__menuitem--primary",
      "onClick": [Function],
      "to": "linkyLink",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "",
          "color": "",
          "name": "menu-default-interface",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "title",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "menu__menuitem--active",
        "ariaCurrent": "true",
        "children": Array [
          <Icon
            className=""
            color=""
            name="menu-default-interface"
            style={Object {}}
/>,
          "title",
        ],
        "className": "menu__menuitem menu__menuitem--primary",
        "onClick": [Function],
        "to": "linkyLink",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "",
            "color": "",
            "name": "menu-default-interface",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "title",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
